ingestion/monitoring/completeness.py

```diff
@@ -53,14 +56,23 @@ def completeness_s3(s3_object, endpoint_url=None):
         s3_object.bucket_name
     )
     with tempfile.NamedTemporaryFile() as tmp:
+        tries = 0
+        success = False
         logging.info(f"Downloading {s3_object.key} from {bucket}")
-        try:
-            bucket.download_file(s3_object.key, tmp.name)
-        except Exception as e:
-            logging.error(f"Failed to download with exception {e}")
+        for tries in range(MAX_TRIES):
+            try:
+                bucket.download_file(s3_object.key, tmp.name)
+                success = True
+                break
+            except Exception:
+                logging.info(f"Failed to download {s3_object}, retrying in {2**tries} seconds ...")
```

Comment:
could probably look at the exception to decide whether it's temporary (worth retrying) or not, but definitely should still log it.

Reply:
What was the exception that led to adding `sleep`?